newIDE/app/src/MainFrame/index.js

```diff
@@ -409,6 +409,24 @@ class MainFrame extends React.Component<Props, State> {
       .then(({ content, fileMetadata }) => {
         const serializedProject = gd.Serializer.fromJSObject(content);
 
+        if(!content.gdVersion && content.eventsFunctions){
```

Comment:
Would it be possible to move this to a function in new file? 
Probably a file called "ProjectContentChecker" in "src/ProjectStorage". You can name the function "verifyProjectContent" and pass `(i18n, content)` as arguments.

(In this new file, you'll need to add flow comment at the top, and import showErrorBox, t and I18nType)

Also you would need to call this method before `const serializedProject = gd.Serializer.fromJSObject(content);`, because this method is creating a new object in libGD.js memory. If `serializedProject.delete()` is not called, this is creating a memory leak (serializedProject will stay forever in memory).